controllers/hospital.controller.js

```diff
@@ -0,0 +1,109 @@
+const Hospital = require("../models/hospital.model");
+
+
+
+exports.addHospital = async (req, res) => {
+
+
+
+    const hospitalObj = {
+        name: req.body.name,
+        address: req.body.address,
+        doctor_ids: []
+    }
+
+    try {
+        const hospital = await Hospital.create(hospitalObj);
+        console.log(hospital)
+
+        return res.status(201).send(hospital);
+
+    } catch (err) {
+        console.log(err.message);
+        return res.status(500).send({
+            message: "Some internal error"
+        })
+    }
+
+}
+
+exports.getAllHospital = async (req, res) => {
+    try {
+        const hospital = await Hospital.find();
+        return res.status(200).send(hospital)
+    } catch (err) {
+        console.log(err.message);
+        return res.status(500).send({
+            message: "Some internal error"
+        })
+    }
+    
+}
+
+exports.getOneHospital = async (req, res) => {
+    try{
+            
+        const hospital = await Hospital.findOne({
+            _id: req.params.id
+        });
+
+        res.status(200).send(hospital);
+    }catch(err){
+        console.log(err.message);
+        return res.status(500).send({
+            message: "Some internal error"
+        })
+    }
+}
+
+exports.updateHospital = async (req, res) => {
+
+    try{
+        const hospital = await Hospital.findOne({
+            _id: req.params.id
+        });
+    
+        console.log(hospital);
+    
+        if (hospital == null) {
+            return res.status(200).send({
+                message: "Hospital doesn't exist"
+            })
+        }
+    
+    
+    
+        hospital.name = req.body.name != undefined ? req.body.name : hospital.name;
+        hospital.address = req.body.address != undefined ? req.body.address : hospital.address;
+        hospital.doctor_ids = req.body.doctor_ids != undefined ? req.body.doctor_ids : hospital.doctor_ids;
```

Comment:
this will replace the list of the doctors.

We can improve this to support the removal or addition of a new doctor in hospital